### PR TITLE
Adding support for KV-only clusters

### PR DESCRIPTION
**Pull request by deniswsrosa**
- If the cluster is KV we will use a different way of listing and showing metadata of documents
- Counters are now read from the KV node instead of running counters (muuuuch more efficient)
- Fixed the issue with folding
- Show/hide features if the Query Service is enabled.
- Upgraded the jetbrains version to 2022.3.3
- fixed some issues with deprecated methods

### CODE REVIEW

File: src/main/java/com/couchbase/intellij/database/ActiveCluster.java [hunk 1]
```diff
@@ -77,12 +85,15 @@ public void connect(SavedCluster savedCluster, Runnable disconnectListener) {
 
         try {
             String password = DataLoader.getClusterPassword(savedCluster);
-            ClusterEnvironment env = ClusterEnvironment.builder().build();
 
             cluster = Cluster.connect(savedCluster.getUrl(),
-                    ClusterOptions.clusterOptions(savedCluster.getUsername(), password).environment(env));
+                    ClusterOptions.clusterOptions(savedCluster.getUsername(), password).environment(env -> {
+                        //env.applyProfile("wan-development");
```
Comment (deniswsrosa) [inline src/main/java/com/couchbase/intellij/database/ActiveCluster.java:91]: this is a problem btw, I will report a bug for you.

File: src/main/java/com/couchbase/intellij/database/ActiveCluster.java [hunk 2]
```diff
@@ -94,14 +105,37 @@ public void connect(SavedCluster savedCluster, Runnable disconnectListener) {
...
                 this.color = Color.decode(savedCluster.getColor());
             }
+
+            ServerOverview overview = CouchbaseRestAPI.getOverview();
+            setServices(overview.getNodes().stream()
+                    .flatMap(node -> node.getServices().stream()).distinct().collect(Collectors.toList()));
+
+            setVersion(overview.getNodes().get(0).getVersion()
+                    .substring(0, overview.getNodes().get(0).getVersion().indexOf('-')));
+
```
Comment (deniswsrosa) [inline src/main/java/com/couchbase/intellij/database/ActiveCluster.java:125]: It is mandatory to get the available services before saying that we connected to the cluster

File: src/main/java/com/couchbase/intellij/database/ActiveCluster.java [hunk 3]
```diff
@@ -94,14 +105,37 @@ public void connect(SavedCluster savedCluster, Runnable disconnectListener) {
...
+            //Notify Listeners that we connected to a new cluster.
+            //NOTE: Only singletons can register here, otherwise we will get a memory leak
+            CompletableFuture.runAsync(() -> {
+                for(Runnable run: newConnectionListener) {
+                    try {
```
Comment (deniswsrosa) [inline src/main/java/com/couchbase/intellij/database/ActiveCluster.java:130]: this is required to notify the toolwindows that we connected to a new cluster.

File: src/main/java/com/couchbase/intellij/tree/TreeRightClickListener.java [hunk 4]
```diff
@@ -407,32 +418,35 @@ private static void handleCollectionRightClick(Project project, MouseEvent e, De
...
-            JBMenuItem clearDocFilter = new JBMenuItem("Clear Document Filter");
-            popup.add(clearDocFilter);
-            clearDocFilter.addActionListener(e12 -> {
-                QueryFiltersStorage.getInstance().getValue().saveQueryFilter(ActiveCluster.getInstance().getId(), col.getBucket(), col.getScope(), col.getText(), null);
 
-                col.setQueryFilter(null);
-                TreePath treePath = new TreePath(clickedNode.getPath());
-                tree.collapsePath(treePath);
-                tree.expandPath(treePath);
+        if (ActiveCluster.getInstance().hasQueryService()) {
```
Comment (deniswsrosa) [inline src/main/java/com/couchbase/intellij/tree/TreeRightClickListener.java:423]: I just added this if here and reformatted the whole thing

File: src/main/java/com/couchbase/intellij/workbench/QueryResultToolWindowFactory.java [hunk 5]
```diff
@@ -138,7 +143,9 @@ public void actionPerformed(@NotNull AnActionEvent e) {
         topPanel.add(toolbar.getComponent(), BorderLayout.EAST);
 
 
-        Document document = EditorFactory.getInstance().createDocument("{\n\"No data to display\": \"Hit 'execute' in the query editor to run a statement.\"\n}");
+        VirtualFile virtualFile = new LightVirtualFile("query_result", FileTypeManager.getInstance().getFileTypeByExtension("json"),
```
Comment (deniswsrosa) [inline src/main/java/com/couchbase/intellij/workbench/QueryResultToolWindowFactory.java:146]: This line fixes the issue with the folding

**Review by chedim — APPROVED** [review]

**Pull request merged**